### PR TITLE
events update

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://bugzilla.redhat.com/show_bug.cgi?id=1484537

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
@tnguyen-rh This one is for https://bugzilla.redhat.com/show_bug.cgi?id=1484537 PTAL

Can I get some help with this one?

**Review by knobunc — APPROVED** [review]
The networking events LGTM.

/lgtm

**Review by pravisankar — COMMENTED** [review]
Some more events might be missing here. I do not see Scheduler, Pod auto scale events.

**Comment by pravisankar** [comment]
Scheduler Events:
---------------------
FailedScheduling  :  Failed to schedule pod: `<pod-namespace>/<pod-name>` 
This event is raised for various reasons: AssumePodVolumes failed, Binding rejected, etc.

Preempted  :  By `<preemptor-namespace>/<preemptor-name>` on node `<node-name>`

Scheduled  :  Successfully assigned `<pod-name>` to `<node-name>`

**Comment by pravisankar** [comment]
Horizontal Pod AutoScaler Events:
--------------------------------------

SelectorRequired  :  Selector is required

InvalidSelector  :  Couldn't convert selector into a corresponding internal selector object

FailedGetObjectMetric  :  HPA was unable to compute the replica count

InvalidMetricSourceType  :  Unknown metric source type

ValidMetricFound  :  HPA was able to successfully calculate a replica count

FailedConvertHPA  :  Failed to convert the given HPA

FailedGetScale  :   HPA controller was unable to get the target's current scale

SucceededGetScale  :  HPA controller was able to get the target's current scale

FailedComputeMetricsReplicas  :  Failed to compute desired number of replicas based on listed metrics

FailedRescale  :  New size: `<size>`; reason: `<msg>`; error: `<error-msg>`

SuccessfulRescale  :  New size: `<size>`; reason: `<msg>`

FailedUpdateStatus  :  Failed to update status

**Comment by gaurav-nelson** [comment]
Thank you @pravisankar 
I will add these ones as well.

**Comment by pravisankar** [comment]
DaemonSet Events:
-----------------------

SelectingAll  :  This daemon set is selecting all pods. A non-empty selector is required

FailedPlacement  :  Failed to place pod on `<node-name>`

FailedDaemonPod  :  Found failed daemon pod `<pod-name>` on node `<node-name>`, will try to kill it

**Comment by pravisankar** [comment]
LoadBalancer Events:
------------------------

CreatingLoadBalancerFailed  :  Error creating load balancer

DeletingLoadBalancer  :  Deleting load balancer

EnsuringLoadBalancer  :  Ensuring load balancer

EnsuredLoadBalancer  :  Ensured load balancer

UnAvailableLoadBalancer :  There are no available nodes for LoadBalancer service

LoadBalancerSourceRanges  :  `<old-source-range>` -> `<new-source-range>`

LoadbalancerIP  :  `<old-ip>` -> `<new-ip>`

ExternalIP  :  Added: `<external-ip>`

UID  : `<old-service-uid>` -> `<new-service-uid>`

ExternalTrafficPolicy :  `<old-policy>` -> `<new-ploicy>`

HealthCheckNodePort  : `<old-node-port>` -> `new-node-port>`

UpdatedLoadBalancer  :  Updated load balancer with new hosts

LoadBalancerUpdateFailed  :  Error updating load balancer with new hosts

DeletingLoadBalancer  :  Deleting load balancer

DeletingLoadBalancerFailed  :  Error deleting load balancer

DeletedLoadBalancer  :  Deleted load balancer

**Comment by gaurav-nelson** [comment]
Thanks @pravisankar I have added all these event as well. I am now working on their descriptions and will soon put this up for a review.

**Comment by gaurav-nelson** [comment]
[[DOCS] Add documentation for ALL events](https://trello.com/c/yRIZC72I/1-docs-add-documentation-for-all-events)

File: dev_guide/events.adoc [hunk 1]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+|HPA controller was unable to get the target's current scale
+
+|`*SucceededGetScale*`
+|HPA controller was able to get the target's current scale
+
+|`*FailedComputeMetricsReplicas*`
+|Failed to compute desired number of replicas based on listed metrics
+
+|`*FailedRescale*`
+|New size: <size>; reason: <msg>; error: <error-msg>
```
Comment (pravisankar) [inline dev_guide/events.adoc]: New size: `<size>`; reason: `<msg>`; error: `<error-msg>`

File: dev_guide/events.adoc [hunk 2]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+
+|`*SuccessfulRescale*`
+|New size: <size>; reason: <msg>
```
Comment (pravisankar) [inline dev_guide/events.adoc]: New size: `<size>`; reason: `<msg>`

File: dev_guide/events.adoc [hunk 3]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+|Lists external IP address. For example, `Added: <external-ip>`.
+
+|`*UID*`
+|Lists the new UID. For example, `<old-service-uid> -> <new-service-uid>`.
+
+|`*ExternalTrafficPolicy*`
+|Lists the new `ExternalTrafficPolicy`. For example, `<old-policy> -> <new-ploicy>`
+
+|`*HealthCheckNodePort*`
+|Lists the new `HealthCheckNodePort. For example, `<old-node-port> -> new-node-port>`
```
Comment (pravisankar) [inline dev_guide/events.adoc]: `HealthCheckNodePort`

**Review by pravisankar — CHANGES_REQUESTED** [review]

**Review by pravisankar — APPROVED** [review]
/lgtm

**Comment by gaurav-nelson** [comment]
@bmeng Please take a look

**Comment by gaurav-nelson** [comment]
@openshift/team-documentation PTAL

File: dev_guide/events.adoc [hunk 4]
```diff
@@ -77,6 +77,13 @@ This section describes the events of {product-title}.
 
 |`*Started*`
 |Container started.
+
+|`*Preempting*`
+|Preempting other pods
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop after sentence?

File: dev_guide/events.adoc [hunk 5]
```diff
@@ -173,6 +180,70 @@ This section describes the events of {product-title}.
...
+|Enforced System Reserved Cgroup limit.
+
+|`*UnsupportedMountOption*`
+|Unsupported mount option.
+
+|`*SandboxChanged*`
+|Pod sandbox changed.
+
+|`*FailedCreatePodSandBox*`
+|Failed create pod sandbox.
```
Comment (bfallonf) [inline dev_guide/events.adoc]: "Failed to create pod sandbox" maybe?

File: dev_guide/events.adoc [hunk 6]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+| Name | Description
+
+|`*FailedKillPod*`
+|Failed to stop a pod.
+
+|`*FailedCreatePodContainer*`
+|Failed to create a pod contianer.
+
+|`*Failed*`
+|Failed to make pod data directories
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop?

File: dev_guide/events.adoc [hunk 7]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+
+|===
+
+.Horizontal Pod AutoScaler Events
+[cols="2,8",options="header"]
+|===
+| Name | Description
+
+|SelectorRequired
+|Selector is required
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop?

File: dev_guide/events.adoc [hunk 8]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+
+|`*InvalidSelector*`
+|Couldn't convert selector into a corresponding internal selector object
```
Comment (bfallonf) [inline dev_guide/events.adoc]: s/couldn't/could not

File: dev_guide/events.adoc [hunk 9]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+
+|`*FailedGetObjectMetric*`
+|HPA was unable to compute the replica count
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop? For the rest of this table as well.

File: dev_guide/events.adoc [hunk 10]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+|Occurs when pod is recycled.
+
+|`*VolumeDelete*`
+|Occurs when volume is deleted.
+
+|`*VolumeFailedDelete*`
+|Error when deleting the volume.
+
+|`*ExternalProvisioning*`
+|Occurs when volume for the claim is provisioned either manually or via external software
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop? And on the one below the one below as well.

File: dev_guide/events.adoc [hunk 11]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+| Name | Description
+
+|`*FailedPostStartHook*`
+|Handler failed for pod start.
+
+|`*FailedPreStopHook*`
+|Handler failed for pre stop.
+
+|`*UnfinishedPreStopHook*`
+|Pre stop hook unfinished.
```
Comment (bfallonf) [inline dev_guide/events.adoc]: s/pre stop/pre-stop

File: dev_guide/events.adoc [hunk 12]
```diff
@@ -192,3 +263,263 @@ This section describes the events of {product-title}.
...
+.DaemonSet Events
+[cols="2,8",options="header"]
+|===
+| Name | Description
+
+|`*SelectingAll*`
+|This daemon set is selecting all pods. A non-empty selector is required.
+
+|`*FailedPlacement*`
+|Failed to place pod on `<node-name>`
```
Comment (bfallonf) [inline dev_guide/events.adoc]: Full stop

**Comment by bfallonf** [comment]
@gaurav-nelson I made a few comments. Mostly full stop is missing-related.

**Comment by gaurav-nelson** [comment]
Thanks @bfallonf I have fixed those. Merging this one now.

**Comment by gaurav-nelson** [comment]
Cherry-picked in 3.5 branch as well, because the original bug was against version 3.5

**Pull request merged**